app/components/pageContent/index.tsx

```diff
@@ -1,48 +1,51 @@
 import { FC } from "react";
-import SearchBar from "../searchBar";
-import CardList, { CardsData } from "../cardList";
-import { ResponsiveDeckWrapper } from "../responsiveDeckWrapper/ResponsiveDeckWrapper";
-import Deck from "../deck/deck";
+import { CardsData } from "../cardList";
+import { PageTypes } from "../../components/providers/cardDataProvider";
 import { DeckListProvider } from "../providers/deckListProvider/DeckListProvider";
-import classNames from "classnames";
-import { SaveButton } from "../saveButton";
+import { CommonPageContent } from "./commonPageContent";
 
 export type PageContentProps = {
   cardList: CardsData[];
   id: string;
-  saveType: string;
   saveCollection?: Function;
+  userId?: string;
+  deckLists?: string[];
+  pageType?: string;
 };
 
 export const PageContent: FC<PageContentProps> = ({
   cardList,
   id,
-  saveType,
   saveCollection,
+  userId,
+  deckLists,
+  pageType,
 }) => {
-  return (
-    <div>
-      <DeckListProvider id={id}>
-        <SearchBar></SearchBar>
-        <div className={"flex p-8"}>
-          <CardList collectionView={true} cardArray={Object.values(cardList)} />
-          <ResponsiveDeckWrapper>
-            <div
-              className={classNames(
-                "bg-gb-brown rounded-md overflow-hidden",
-                "h-full w-full shadow-2xl shadow-[rgba(0,0,0,0.35)]",
-                "border-4 border-[rgb(224,180,34)]"
-              )}
-            >
-              <Deck isCollection={false} />
-            </div>
-            <SaveButton
-              saveType={saveType}
-              saveCollection={saveCollection}
-            ></SaveButton>
-          </ResponsiveDeckWrapper>
-        </div>
-      </DeckListProvider>
-    </div>
-  );
+  if (pageType === PageTypes.DECKBUILDER) {
+    return (
+      <div>
+        <DeckListProvider id={id}>
+          <CommonPageContent
+            cardList={cardList}
+            saveType={pageType}
+            saveCollection={saveCollection}
+            userId={userId}
+            deckLists={deckLists}
+          />
+        </DeckListProvider>
```

Comment:
Nothing inherently wrong with putting the `DecklistProvider` here; just curious if there was a specific reason to put it here instead of with the other providers

Reply:
NVM I think I see your reasoning